src/QsCompiler/QirGeneration/EntryPointOperationLoader.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Quantum.QsCompiler.QIR.BondSchemas;
+using Microsoft.Quantum.QsCompiler.SyntaxTree;
+
+namespace Microsoft.Quantum.QsCompiler.QIR
+{
+    public static class EntryPointOperationLoader
+    {
+        public static IList<EntryPointOperation> LoadEntryPointOperations(FileInfo assemblyFileInfo)
+        {
+            if (!AssemblyLoader.LoadReferencedAssembly(assemblyFileInfo.FullName, out var compilation))
+            {
+                throw new ArgumentException("Unable to read the Q# syntax tree from the given DLL.");
+            }
+            return GenerateEntryPointOperations(compilation);
+        }
+
+        private static IList<EntryPointOperation> GenerateEntryPointOperations(QsCompilation compilation)
+        {
+            return compilation.EntryPoints.Select(ep => new EntryPointOperation()
+            {
+                Name = NameGeneration.InteropFriendlyWrapperName(ep)
```

Comment:
This method should also populate the `Parameters` list for each entry-point.

Reply:
I've populated the Parameters such that each parameter has a name (as it appears in Q#) and a type. I've had to write my own function for mapping the Q# types to the QIR types as I couldn't find anything else that would do this, but it seems kind of awkward. Let me know if there is a better way to do the type mapping or if there is something somewhere that does this already that I can use here.

Reply:
Resolving since this has been addressed.